contracts/contracts/subnet/SubnetActorManagerFacet.sol

```diff
@@ -177,31 +177,56 @@ contract SubnetActorManagerFacet is SubnetActorModifiers, ReentrancyGuard, Pausa
         }
     }
 
-    function _patchGenesisValidators(address validator) internal {
+    function _increaseGenesisValidatorCollateral(address validator, uint256 amount) internal {
         // add to initial validators avoiding duplicates if it
         // is a genesis validator.
         bool alreadyValidator;
         uint256 length = s.genesisValidators.length;
-        for (uint256 i; i < length; ) {
+        uint256 i;
+        for (; i < length; ) {
             if (s.genesisValidators[i].addr == validator) {
-                alreadyValidator = true;
                 break;
             }
             unchecked {
                 ++i;
             }
         }
-        if (!alreadyValidator) {
+        // already validator if we iterate through all of them but didn't get a match
+        if (i >= length) {
+            s.genesisValidators[i].weight += amount;
```

Comment:
This is incorrect right? Give we break early when found then `i < length` means found.

Reply:
Yeah, exactly. By definition we did not find a validator key if we reach `i == length` which is the loop abort condition, so we can use `i >= length` as a defensive way of checking `none found` vs `some found`.